pints/_mcmc/__init__.py

```diff
@@ -600,6 +623,34 @@ def run(self):
         # Start sampling
         timer = pints.Timer()
         running = True
+
+        # initialisation (for single chain methods only and only if prior
```

Comment:
I'm not super excited about sticking more stuff in the controller. It's a very complex class already, and this is a really easy chunk to break off and do externally?

Reply:
I think we should add this because it's a convenience function that is really useful in practice. Chon himself said it would be useful because it would save him having to take care of this. Otherwise, it makes people lazy -- we currently are initialising wrongly (as per best practices) in about 95% of our notebooks; I don't feel comfortable about this. I suspect if we don't rectify this that it will come up in review.

Reply:
I do agree that the run() function of the controller is complicated too and, as I've said before, I would like to simplify it by splitting up into separate methods. But that's for another day.

Reply:
Most of the things in `run` can't be isolated (I don't agree with that tool's analysis!), but this bit easily can. Maybe stick it in a hidden method inside then controller then and just call that?
e.g.
```
def __init__(self, x0, ...):
    if isinstance(x0, pints.LogPrior):
        self._log_prior = x0
        self._x0 = None
    elif len(x0) etc.

def run(..)
    ...
    if self._x0 is None:
        self._sample_x0()
    ...
def _sample_x0():
    set x0 based on log_prior passed at construction time
```

Reply:
Cool -- thanks, will do.